homeassistant/components/switcher_kis/config_flow.py

```diff
@@ -2,9 +2,97 @@
 
 from __future__ import annotations
 
-from homeassistant.helpers import config_entry_flow
+from collections.abc import Mapping
+from typing import Any
 
-from .const import DOMAIN
-from .utils import async_has_devices
+import voluptuous as vol
 
-config_entry_flow.register_discovery_flow(DOMAIN, "Switcher", async_has_devices)
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+
+from .const import CONF_TOKEN, CONF_USERNAME, DATA_DISCOVERY, DOMAIN
+from .utils import async_discover_devices
+
+
+class SwitcherFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle Switcher config flow."""
+
+    VERSION = 1
+
+    entry: ConfigEntry | None = None
+    username: str | None = None
+    token: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the start of the config flow."""
+        if self._async_current_entries(True):
+            return self.async_abort(reason="single_instance_allowed")
+
+        self.hass.data.setdefault(DOMAIN, {})
+        if DATA_DISCOVERY not in self.hass.data[DOMAIN]:
+            self.hass.data[DOMAIN][DATA_DISCOVERY] = self.hass.async_create_task(
```

Comment:
This is already async, why does it run in a task?

Reply:
I just copied it over from the previous iteration of this file..

Reply:
You copied parts which are not related and mixed them, can  you remove the `async_create_task`?

Reply:
I revert that part back.. I dont know how to progress from here..